stratum/hal/lib/tdi/tdi_sde_interface.h

```diff
@@ -327,10 +329,13 @@ class TdiSdeInterface {
   // TODO(max): figure out optional register index API, see TotW#163
   virtual ::util::Status ReadIndirectMeters(
       int device, std::shared_ptr<TdiSdeInterface::SessionInterface> session,
-      uint32 table_id, absl::optional<uint32> meter_index,
-      std::vector<uint32>* meter_indices, std::vector<uint64>* cirs,
-      std::vector<uint64>* cbursts, std::vector<uint64>* pirs,
-      std::vector<uint64>* pbursts, std::vector<bool>* in_pps) = 0;
+      uint32 table_id, absl::optional<int64> meter_index,
+      std::vector<int64>* meter_indices, std::vector<int64>* cirs,
+      std::vector<int64>* cbursts, std::vector<int64>* pirs,
+      std::vector<int64>* pbursts, std::vector<int64>* cirs_unit,
+      std::vector<int64>* cbursts_unit, std::vector<int64>* pirs_unit,
```

Comment:
I think all the code added in hal/lib/tdi in this PR needs to be moved to hal/lib/tdi/es2k since none of them are applicable to other TDI platforms and are es2k specific. We should think about overriding functions if needed

Reply:
`TdiSdeInterface` is an abstract interface. `TdiSdeWrapper` is the concrete class. The methods must be defined in the abstract interface if they are to be invoked from common code. We can provide dummy implementations at the TdiSdeWrapper level and actual implementations for ES2K.

Reply:
1. Not breaking other targets is high on my list of priorities.
2. So is not breaking other users of Stratum.
3. Source backward compatibility is important. Components are not updated in lock step.